src/uu/rm/src/rm.rs

```diff
@@ -575,24 +611,141 @@ fn is_writable(_path: &Path) -> bool {
     true
 }
 
+#[cfg(unix)]
+fn should_skip_different_device(
```

Comment:
this function should be removed 
should_skip_different_device_with_dev should be called directly